test/e2e/tests/health_check.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build e2e
+// +build e2e
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+	gwv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gwv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HealthCheckTest)
+}
+
+var HealthCheckTest = suite.ConformanceTest{
```

Comment:
can we rename this into `DownstreamHealthCheck`? Since this health check is in CTP, as we also have health check in BTP, and the `UpstreamHealthCheck` e2e test case will be added soon.

Reply:
+1 prefer `ListenerHealthCheck`